src/hotspot/share/gc/shenandoah/heuristics/shenandoahOldHeuristics.cpp

```diff
@@ -504,15 +548,36 @@ bool ShenandoahOldHeuristics::should_start_gc() {
 
   if (_fragmentation_trigger) {
     ShenandoahHeap* heap = ShenandoahHeap::heap();
+    const size_t region_size_bytes = ShenandoahHeapRegion::region_size_bytes();
+
     size_t used = _old_generation->used();
     size_t used_regions_size = _old_generation->used_regions_size();
+
+    // used_regions includes humongous regions
     size_t used_regions = _old_generation->used_regions();
     assert(used_regions_size > used_regions, "Cannot have more used than used regions");
+
+    size_t first_active_region = heap->free_set()->first_old_region();
+    size_t last_active_region = heap->free_set()->last_old_region();
+    size_t span_of_active_regions = (last_active_region > first_active_region)? last_active_region + 1 - first_active_region: 0;
```

Comment:
Thanks for this catch also.  I went through a few iterations of how to implement this, and forgot to clean up all of my experiments.  Am removing this code and am removing first_old_region() and last_old_region().  We only "need" the first and last old region info immediately after we rebuild the Shenandoah Free Set.  Since the rebuild process already iterates over all regions, it's best to compute the information during that pass.